draft-liu-teas-transport-network-slice-yang.txt

```diff
@@ -0,0 +1,1568 @@
+   license-info) in effect on the date of publication of this document.
+   Please review these documents carefully, as they describe your rights
+   and restrictions with respect to this document.  Code Components
+   extracted from this document must include Revised BSD License text as
+   described in Section 4.e of the Trust Legal Provisions and are
+   provided without warranty as described in the Revised BSD License.
+
+Table of Contents
+
+   1.  Introduction  . . . . . . . . . . . . . . . . . . . . . . . .   3
+     1.1.  Terminology . . . . . . . . . . . . . . . . . . . . . . .   3
+     1.2.  Tree Diagram  . . . . . . . . . . . . . . . . . . . . . .   4
+     1.3.  Prefixes in Data Node Names . . . . . . . . . . . . . . .   4
+   2.  Modeling Considerations . . . . . . . . . . . . . . . . . . .   4
+     2.1.  Relationships to Related Topology Models  . . . . . . . .   4
+     2.2.  ACTN for Network Slicing  . . . . . . . . . . . . . . . .   5
+   3.  Model Applicability . . . . . . . . . . . . . . . . . . . . .   5
+   4.  YANG Model Overview . . . . . . . . . . . . . . . . . . . . .   8
+   5.  Model Tree Structure  . . . . . . . . . . . . . . . . . . . .   8
+   6.  YANG Modules  . . . . . . . . . . . . . . . . . . . . . . . .  11
+   7.  Manageability Considerations  . . . . . . . . . . . . . . . .  18
+   8.  Security Considerations . . . . . . . . . . . . . . . . . . .  19
+   9.  IANA Considerations . . . . . . . . . . . . . . . . . . . . .  19
+   10. References  . . . . . . . . . . . . . . . . . . . . . . . . .  20
+     10.1.  Normative References . . . . . . . . . . . . . . . . . .  20
+     10.2.  Informative References . . . . . . . . . . . . . . . . .  21
+   Appendix A.  Acknowledgments  . . . . . . . . . . . . . . . . . .  22
+   Appendix B.  Data Tree for the Example in Section 3.1 . . . . . .  22
+     B.1.  Native Topology . . . . . . . . . . . . . . . . . . . . .  23
+   Authors' Addresses  . . . . . . . . . . . . . . . . . . . . . . .  27
+
+
+
+
+
+
+
+Liu, et al.              Expires 13 August 2023                 [Page 2]
+
+Internet-Draft      Network Slice Topology Data Model      February 2023
+
+
+1.  Introduction
+
+   This document defines a YANG [RFC7950] data model for representing,
+   managing, and controlling IETF network slices as abstract network
+   topologies, where the network slices are defined in
+   [I-D.ietf-teas-ietf-network-slices].
+
+   The defined data model is an interface between customers and
+   providers for configurations and state retrievals, so as to support
+   network slicing as a service.  Through this model, a customer can
+   learn the slicing capabilities and the available resources of the
+   provider.  A customer can request or negotiate with a network slicing
+   provider to create an instance.  The customer can incrementally
+   update its requirements on individual topology elements in the slice
+   instance, and retrieve the operational states of these elements.
```

Comment:
customer rquest: A --- B --- C (100G) 
response: A --- B --- C (50G negotiated)
later: A --- B --- C --- D  (50G)
response: A --- B --- C --- D (50G negotiated) 
topology configuration = true, topology state may be used as hints towards the customer regarding the negotiation
Add the above as an example to the text.

Reply:
Since this paragraph is in the introduction section, it is not easy to add a detailed description of the example. Instead, I'd propose the following "e.g." text:
The customer can incrementally update its requirements on individual topology elements in the slice instance, e.g., adding or removing a node or link, updating desired bandwidth of a link, and retrieve the operational states of these elements.